# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/Stencil.cpp
# Commit: 11fc8f557a2a
# Full Hash: 11fc8f557a2a7cf3c89e7c1396316aabd9725778
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 6 - Move Realm* from ObjectGroup to BaseShape. r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106976
# ==============================================================================

diff -r 9bfa6023a553 -r 11fc8f557a2a js/src/frontend/Stencil.cpp
--- a/js/src/frontend/Stencil.cpp	Sat Mar 06 01:05:20 2021 +0000
+++ b/js/src/frontend/Stencil.cpp	Sat Mar 06 01:05:21 2021 +0000
@@ -927,9 +927,10 @@
   if (!group) {
     return false;
   }
-  RootedShape shape(cx, EmptyShape::getInitialShape(
-                            cx, &JSFunction::class_, TaggedProto(proto),
-                            /* nfixed = */ 0, ObjectFlags()));
+  RootedShape shape(
+      cx, EmptyShape::getInitialShape(cx, &JSFunction::class_, cx->realm(),
+                                      TaggedProto(proto),
+                                      /* nfixed = */ 0, ObjectFlags()));
   if (!shape) {
     return false;
   }